pkg/roll/run_update.go

```diff
@@ -256,17 +258,30 @@ func (u update) countHealthy(id rcf.ID, checks map[string]health.Result) (rcNode
 	if err != nil {
 		return ret, err
 	}
-	ret.desired = rcFields.ReplicasDesired
+	ret.Desired = rcFields.ReplicasDesired
 
 	nodes, err := rc.New(rcFields, u.kps, u.rcs, u.sched, u.labeler, u.logger).CurrentNodes()
 	if err != nil {
 		return ret, err
 	}
-	ret.current = len(nodes)
+	ret.Current = len(nodes)
 
 	for _, node := range nodes {
+		// TODO: is reality checking an rc-layer concern?
```

Comment:
I might end up moving it to RC anyway, but I'm not sure I agree with your argument - the functionality of pkg/replication is partially replaced by RUs and partially by RCs. RCs alone aren't equivalent to pkg/replication and in particular they have no safety guarantees when their replica count is changed. That's why I put this in RUs originally (because I felt that the safety-aware code belonged there).